play/src/lib/HelperFunctions.tsx

```diff
@@ -85,6 +86,50 @@ export const checkForSubmittedAnswerOnRejoin = (
   return { selectedAnswerIndex, isSubmitted };
 };
 
+/**
+ * on rejoining game, this checks if the player has already selected a confidence level
+ * @param hasRejoined - if a player is rejoining
+ * @param currentAnswer - the player's answer to the current question
+ * @param currentState - the current state of the game session
+ * @returns - the index of the confidence the player has submitted, null if they haven't selected a confidence, boolean to track submission
+ */
+export const checkForSelectedConfidenceOnRejoin = (
+  hasRejoined: boolean,
+  currentAnswer: ITeamAnswer | null | undefined,
+  currentState: GameSessionState
+): {
+  selectedConfidenceIndex: number | null;
+  isSelected: boolean;
+  timeOfLastSelect: number | null;
+} => {
+  let selectedConfidenceIndex = null;
+  let isSelected = false;
+  // here, since we do not store time of last select in the backend (5 seconds would be negligible on refresh),
+  // we set the timeOfLastSelect to null to re-initialize the value instead of populating it with the previous value
+  const timeOfLastSelect = null;
+  // adding dictionary to account for string casting for material UI components
+  const confidenceOptions = [
+    ConfidenceLevel.NOT_AT_ALL,
+    ConfidenceLevel.KINDA,
+    ConfidenceLevel.QUITE,
+    ConfidenceLevel.VERY,
+    ConfidenceLevel.TOTALLY,
+  ];
+  if (
+    hasRejoined &&
+    (currentState === GameSessionState.CHOOSE_CORRECT_ANSWER ||
+      currentState === GameSessionState.CHOOSE_TRICKIEST_ANSWER)
+  ) {
+    if (!isNullOrUndefined(currentAnswer)) {
+      isSelected = currentAnswer.confidenceLevel !== ConfidenceLevel.NOT_RATED;
+      selectedConfidenceIndex = confidenceOptions.indexOf(
+        currentAnswer.confidenceLevel
+      );
+    }
+  }
+  return { selectedConfidenceIndex, isSelected, timeOfLastSelect };
```

Comment:
We should still return the confidence enum itself instead of the index. Returning index is coupling the selected confidence with index of it in the array. If permutation changes or we remove / add new confidence, this can cause an error for index out of bound or incorrectly reports data

Reply:
Alex raised this with me in a meeting so probably poor advice on my end. The annoying thing is that mui radio buttons take a `value` prop that has to be a string, so we're stuck doing some sort of conversion. The initial idea was just to do this conversion as close to the component as possible (immediately before making use of the radio button component) but thinking about it, we could probably just stringify our enum and pass it to the radio button component. I think this would eliminate any discrepancy between the enum we use and the value we pass to the radio button. In other words, we would always be programmatically referencing the enum state so if it changes, the component will catch that change.

Reply:
Hmm, if the conversion is needed to string, why not add it to the enum itself. Enums can have values in typescript, so you can assign string or int values to them. That's a nice feature that can be useful in these cases

Reply:
ahh ok I'll look into this right now but that's much more readable. I was sort of struggling to find a clean way of dealing with the conversions but the above is great advice, will take another look with all of that in mind!

Reply:
@maniramezan I just had a quick question: my understanding is that the enum would be rewritten as follows: 
`enum ConfidenceLevel {
  NOT_RATED = "NOT_RATED"
  NOT_AT_ALL = "NOT_AT_ALL"
  KINDA = "KINDA"
  QUITE = "QUITE"
  VERY = "VERY"
  TOTALLY = "TOTALLY"
}` but would this change be in networking (meaning the backend would have to be updated again) or would this enum simply just be re-written somewhere in play and when the enum is referenced on the frontend, the string conversion is implied?

Reply:
@Alexandra-Kushnirsky hmm interesting, I forgot this enum is in graphql models and we haven't mapped it to a local enum type, right? If yes, then looks like the AWS generated model already has the string values for you, [ConfidenceLevel](https://github.com/rightoneducation/righton-app/blob/02630b0df5eed09ef1700c99fbffb81554d265a4/networking/src/AWSMobileApi.ts#L212). You can also go the re-written path. Latter gives us the opportunity to decouple frontend and backend, but also not necessary at the moment

Reply:
Ok I see, thank you!